app/assets/javascripts/models/page/search_page.js.coffee

```diff
@@ -50,7 +50,7 @@ ns.SearchPage = do (ko
     $('.launch-customize-modal').click ->
       $('#customizeDataModal').modal('show')
 
-    setTimeout initModal, 2000 if !window.edscportal
+    preferences.ready.done(-> initModal()) if !window.edscportal
 
   class SearchPage
     constructor: ->
```

Comment:
>>"The two consecutive calls to preferences will possibly make the code read dirty data if the preferences are changed between the two calls."

They are two calls within the same load procedure (though different scope) - how in the world would any user preference change in that tiny fraction of a time?  That's like suggesting two different reads on a database during load is dangerous as the data might change in between the reads.  

The user preferences won't change if the user does not change them, and it is impossible for a user to change their preferences within the tiny fraction of a second between object instantiations upon page load.